### PR TITLE
feat(core,console): organization mfa requirement

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

configure mfa requirement per organization

image

## Testing

unit and integration tests added

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/oidc/grants/refresh-token.ts [hunk 1]
```diff
@@ -246,6 +246,17 @@ export const buildHandler: (
...
+    // Check if the organization requires MFA and the user has MFA enabled
+    const { isMfaRequired, hasMfaConfigured } = await queries.organizations.getMfaData(
+      organizationId,
+      account.accountId
+    );
+    if (isMfaRequired && !hasMfaConfigured) {
+      const error = new AccessDenied('organization requires MFA but user has no MFA configured');
+      error.statusCode = 403;
+      throw error;
+    }
```
Comment (simeng-li) [inline packages/core/src/oidc/grants/refresh-token.ts:259]: Should we also revoke the current refresh token and session?  e.g. Calling `libraries/user.ts#signoutUser`

File: packages/core/src/queries/organization/index.ts [hunk 2]
```diff
@@ -295,4 +296,32 @@ export default class OrganizationQueries extends SchemaQueries<
...
     super(pool, Organizations);
   }
+
+  /**
+   * Get the multi-factor authentication (MFA) data for the given organization and user.
+   *
+   * @returns Whether MFA is required for the organization and whether the user has configured MFA.
+   * @see {@link MfaData}
+   */
+  async getMfaData(organizationId: string, userId: string) {
```
Comment (simeng-li) [inline packages/core/src/queries/organization/index.ts:306]: Maybe a more specific name like `mfaConfigurationStatus`? mfaData seems too general.

File: packages/core/src/oidc/grants/refresh-token.ts (hunk 1 above)
Comment (gao-sun) [reply packages/core/src/oidc/grants/refresh-token.ts:259]: i don't think so. a user should be available to add MFA and come back to fetch the token afterwards

File: packages/core/src/queries/organization/index.ts (hunk 2 above)
Comment (gao-sun) [reply packages/core/src/queries/organization/index.ts:306]: agreed

**Review by simeng-li — APPROVED** [review]

**Pull request merged**